webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -3,11 +3,11 @@ import { Switch, Route } from 'react-router-dom';
 import { routes } from './routes';
 import { visit } from '../../foreman_navigation';
 
-let currentPath = window.location.pathname;
+let currentPath = window.location.href;
```

Comment:
do we still need this hack?
maybe we can select the location from redux.

Reply:
well, after some deep researching, while I do want to move this workaround away, it requires a persistent store (because redux's store is not persistent while full-page reloading) or at least using `sessionStorage` to keep tracking the previous location. this workaround would be easily fixed by moving to a like-turbolinks ajax fetching, so no full page reload will occur between pages and redux's store keep fresh

Reply:
Thanks for doing the research !
How about maintaining that prevUrl in the window.history:
`window.history.pushState({ prevForemanUrl: window.location.href }, null, "/new/path")`
And than consume it:
`window.history.state.prevForemanUrl`

Reply:
this is another workaround, pretty similar to storing location outside of the component's scope.
I was thinking about changing the fallback route to render a `Legacy` component, which will be responsible for making ajax and in some cases to do a full page reload.
with this solution, we gain a stateful redux across pages, and a performance boost (plus, it's n a react solution :) )
https://github.com/theforeman/foreman/pull/7394